packages/workspace/src/core/hasher/hasher.ts

```diff
@@ -148,12 +162,149 @@ export class Hasher {
     return this.runtimeInputs;
   }
 
+  private tsconfigDepsHash(
+    sortedTransitiveDeps: Array<[string, ProjectGraphNode]>,
+    workspaceJson: any
+  ): Array<{
+    hash: string;
+    record: string;
+  }> {
+    const npmScope = `@${this.nxJson.npmScope}/`;
+    const tsConfig = readTsConfig();
+
+    /**
+     * Parse out the paths from tsconfig so they are not accounted for
+     * in hashing tsconfig configurations since they are hashed separately
+     */
+    const { paths, ...compilerOptions } = tsConfig.compilerOptions;
+    tsConfig.compilerOptions = compilerOptions;
+
+    const hashedTsConfig: {
+      hash: string;
+      record: string;
+    } = {
+      hash: this.hashing.hashObject(sortObject(tsConfig)),
+      record: 'tsconfig.base.json',
+    };
+
+    const hashedTsConfigPaths: {
+      hash: string;
+      record: string;
+    } = {
+      hash: this.hashing.hashArray(
+        sortedTransitiveDeps.map(([name]) => {
+          const project = workspaceJson.projects[name];
+          const path = project.root.replace(
+            project.projectType === 'application' ? 'apps/' : 'libs/',
+            ''
+          );
+          return paths[`${npmScope}${path}`];
+        })
+      ),
+      record: 'tsconfig.base.json[compilerOptions][paths]',
+    };
+
+    return [hashedTsConfig, hashedTsConfigPaths];
+  }
+
+  private workspaceJsonDepsHash(
+    sortedTransitiveDeps: Array<[string, ProjectGraphNode]>,
+    workspaceJson: any,
+    workspaceJsonName: string
+  ): {
+    hash: string;
+    record: string;
+  } {
+    const hashedWorkspaceJsonRecords: string = this.hashing.hashArray(
+      sortedTransitiveDeps.map(([name]) =>
+        this.hashing.hashObject(sortObject(workspaceJson.projects[name]))
+      )
+    );
+
+    return {
+      hash: hashedWorkspaceJsonRecords,
+      record: workspaceJsonName,
+    };
+  }
+
+  /**
+   * Will selectively combine all of the transitive deps of the current
+   * project based on the project graph
+   */
+  async transitiveClosureDepsHash(): Promise<ImplicitHashResult> {
+    const nodes = Object.entries(this.projectGraph.nodes);
+
+    const sortedTransitiveDeps: Array<[
+      string,
+      ProjectGraphNode
+    ]> = nodes
+      .filter(([name, records]) => !name.startsWith('npm:'))
```

Comment:
Wasn't sure if this case always holds true? I am assuming that any node that doesn't start with `npm:` is code that needs to be considered.